### ISSUE TITLE
why do embedded documents inherit schema methods?

### ISSUE THREAD

**Issue by vincentcr** [Automattic/mongoose#3410]
when defining instance methods on a schema, any embedded document schema will also inherit these methods. For example:

```
var mongoose = require('mongoose')
mongoose.connect('mongodb://localhost/test');


var PersonSchema = new mongoose.Schema({
    name: String,
    age: Number,
    creditCard: {
        number: String
    }
})

PersonSchema.methods.sayHello = function() {
    console.log("hello, my name is:", this.name)
}

var Person = mongoose.model('Person', PersonSchema)

var p = new Person({name:'foo', creditCard: { number:'1234'}})

p.sayHello()
p.creditCard.sayHello()

```

This outputs:

```
hello, my name is: foo
hello, my name is: undefined
```

As you can see, the second call "works". It seems to me counterintuitive and even dangerous to have schema methods propagate this way.

**Comment by vkarpov15** [comment]
I'm honestly not sure why this was implemented this way, this behavior predates my involvement. Why do you think this is dangerous?

**Comment by vincentcr** [comment]
After further inspection, it appears that the whole schema is inherited as the embedded document's schema: 

```
console.log('model of credit card:', p.creditCard.constructor.modelName)
console.log('is it the same schema?', p.creditCard.constructor.schema == p.constructor.schema)
```
outputs:

```
model of credit card: Person
is it the same schema? true
```

The trouble, it seems to me, is if you don't know what the object is in advance, and try to inspect its model, you will get a false answer:

In my specific case, I was doing a recursion in a structure containing mongoose objects, and looking for a specific method. the method propagates to the embedded object, pointing to the parent schema, and this breaks the logic.

**Comment by vincentcr** [comment]
So, is this something that you would be open to change?

**Comment by vkarpov15** [comment]
Possibly, really depends on if there's a good reason to. Can you explain to me what the benefit of changing this would be?

**Comment by vkarpov15** [comment]
Was fixed in 4.12 re: #5470
